content/posts/07-meet-mkmr.md

```diff
@@ -0,0 +1,248 @@
+merged into the branch in upstream.
+
+`mkmr` tries to bridge the gap by allowing the user to create the merge request from the
+commandline, by automating pushing a new branch to the user's fork and then creating the
+merge request via the GitLab API.
+
+`mkmr` will prompt before making the merge request, letting you check if your values are
+as expected, this can be skipped by passing `--yes` (and made permanent by passing
+`--save` with it).
+
+![Prompting the user](07-meet-mkmr-02.png)
+
+If one wants to pass a different title, description or labels when creating the merge
+request, one can pass `--edit` (can be made permanent with `--save`), which will open
+`$EDITOR` with instructions, users of `git rebase -i` will notice it is inspired by it.
+
+![Editing the attributes](07-meet-mkmr-03.png)
+
+After checking everything and passing yes to the prompt, the merge request will be created
+
+![Merge Request created](07-meet-mkmr-04.png)
+
+After creating the merge request, the iid of the merge request will be saved in the cache,
+in a file named after the branch of the merge request. This is used by other tools to connect
+a branch name to an specific merge request.
+
+```sh
+$ tree $XDG_CACHE_HOME/mkmr/gitlab.alpinelinux.org/alpine/aports/branches
+/home/enty/var/cache/mkmr/gitlab.alpinelinux.org/alpine/aports/branches
+├── 3.11-busybox
+├── 3.12-alpine-baselayout
+├── 3.12-busybox
+├── arc-theme
+├── busybox
+├── fuse3
+├── perl-module-scandeps
+├── poppler
+├── py3-pytools
+├── py3-validators
+├── screenfetch
+├── strace
+├── tumbler
+├── vips
+└── volumeicon
+
+0 directories, 15 files
+```
+
+### mgmr
+
+The second most important of the `mkmr` tools, `mgmr` handles merging merge requests by
+rebasing them until they are matching upstream, then merging them.
+
+![Mixed args to merge](07-meet-mkmr-06.png)
+
+It can take lots of merge requests and try to merge them all sequentially, it makes its
+own merge train and try to rebase-and-merge each until it succeeds or it hits an an error
+it cannot recover from. It then prints out a summary to the user of what Merge Requests
+were merged and if not what was the error message.
+
+It can take the iid of the Merge Request (the number on the URL one can refer with !N) or
+it can take the name of a branch which will be looked up on the cache and matched to an iid.
+
+![Final Report](07-meet-mkmr-07.png)
+
+After merging it will delete from the cache all branches that were given as they have no
+use for us any longer.
+
+### vimr
+
+This one allows users to edit attributes of a given Merge Request by opening them in `$EDITOR`
+I think it is better shown than talked about as the explanations written to the `$EDITOR` say
+all that needs to be said.
+
+> This is the result of: vimr -n 11361
+
+```null
+# Set to 'close' to close the merge request
+# set to 'reopen' to open a closed merge request
+State: opened
+
+# Line must
```

Comment:
Line must? What does that mean?

Reply:
It means my program has a bug

Reply:
Ok, can you maybe adjust it here to what you change it to in vimr too?

Reply:
Ping